targets/TARGET_STM/qspi_api.c

```diff
@@ -0,0 +1,245 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 3. Neither the name of STMicroelectronics nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if DEVICE_QSPI
+
+#include "qspi_api.h"
+#include "mbed_error.h"
+#include "cmsis.h"
+#include "pinmap.h"
+#include "PeripheralPins.h"
+
+#define QSPI_FLASH_SIZE_DEFAULT 32
+
+void qspi_prepare_command(const qspi_command_t *command, QSPI_CommandTypeDef *st_command) 
+{
+    // TODO: shift these around to get more dynamic mapping
+    switch (command->instruction.bus_width) {
+        case QSPI_CFG_BUS_SINGLE:
+            st_command->InstructionMode = QSPI_INSTRUCTION_1_LINE;
+            break;
+        case QSPI_CFG_BUS_DUAL:
+            st_command->InstructionMode = QSPI_INSTRUCTION_2_LINES;
+            break;
+        case QSPI_CFG_BUS_QUAD:
+            st_command->InstructionMode = QSPI_INSTRUCTION_4_LINES;
+            break;
+    }
+
+    st_command->Instruction = command->instruction.value;
+    st_command->DummyCycles = command->dummy_count,
+    // these are target specific settings, use default values
+    st_command->SIOOMode = QSPI_SIOO_INST_EVERY_CMD;
+    st_command->DdrMode = QSPI_DDR_MODE_DISABLE;
+    st_command->DdrHoldHalfCycle = QSPI_DDR_HHC_ANALOG_DELAY;
+
+    switch (command->address.bus_width) {
+        case QSPI_CFG_BUS_SINGLE:
+            st_command->AddressMode = QSPI_ADDRESS_1_LINE;
+            break;
+        case QSPI_CFG_BUS_DUAL:
+            st_command->AddressMode = QSPI_ADDRESS_2_LINES;
+            break;
+        case QSPI_CFG_BUS_QUAD:
+            st_command->AddressMode = QSPI_ADDRESS_4_LINES;
+            break;
+        default:
+            st_command->AddressMode = QSPI_ADDRESS_1_LINE;
```

Comment:
Use `QSPI_ADDRESS_NONE `instead as default ?

Reply:
I am proposing to change hal to have NONE for bit_width, not in the size. I'll rework this and would be fixed.

For now, I'll set defalt to none.

Reply:
fixed